smelt.lic

```diff
@@ -26,79 +26,63 @@ class Smelt
 
     Flags.add('ingot-done', 'At last the metal appears to be thoroughly mixed')
 
-    get(@rod)
+    DRCC.get_crafting_item(@rod, @bag, @bag_items, @belt)
     if args.refine
-      get('flux')
-      type("pour my flux in crucible", "flux", true)
+      DRCC.get_crafting_item('flux', @bag, @bag_items, @belt)
+      work("pour my flux in crucible", "flux", true)
     else
-      type("stir crucible with my rod", "rod", false)
+      work("stir crucible with my rod", "rod", false)
     end
-    work
-    stow(@rod)
+    DRCC.stow_crafting_item(@rod, @bag, @belt)
   end
 
-  def get(name)
-    DRCC.get_crafting_item(name, @bag, @bag_items, @belt)
-  end
-
-  def stow(name)
-    DRCC.stow_crafting_item(name, @bag, @belt)
-  end
-
-  def type(command, item, stow)
-    @command = command
-    @item = item
-    @stow = stow
-  end
-
-  def work
+  def work(command, item, stow)
     loop do
       break if Flags['ingot-done']
-      case DRC.bput(@command, 'Pour what', 'You can only mix a crucible', 'clumps of molten metal', 'flickers and is unable to consume', 'needs more fuel', 'needs some more fuel', 'roundtime')
+      case DRC.bput(command, 'Pour what', 'You can only mix a crucible', 'clumps of molten metal', 'flickers and is unable to consume', 'needs more fuel', 'needs some more fuel', 'roundtime')
       when 'Pour what'
         DRC.message("Missing Flux")
         break
       when 'You can only mix a crucible'
         break
       when 'clumps of molten metal'
-        stow(@item) if @stow
-        type("turn crucible with my rod", "rod", false)
+        DRCC.stow_crafting_item(@item, @bag, @belt) if DRCI.in_left_hand?(@item)
+        @command = "turn crucible with my rod"
+        @item = "rod"
+        @stow = false
       when 'flickers and is unable to consume'
-        stow(@item) if @stow
-        type("push my bellows", "bellows", true)
-        get('bellows')
+        DRCC.stow_crafting_item(@item, @bag, @belt) if DRCI.in_left_hand?(@item)
+        DRCC.get_crafting_item('bellows', @bag, @bag_items, @belt)
+        @command = "push my bellows"
+        @item = "bellows"
+        @stow = true
       when 'needs more fuel', 'needs some more fuel'
-        stow(@item) if @stow
-        tongs
-        type("push fuel with my #{@item}", @item, true)
+        DRCC.stow_crafting_item(@item, @bag, @belt) if DRCI.in_left_hand?(@item)
+        if @adjustable_tongs && DRCC.get_adjust_tongs('shovel', @bag, @bag_items, @belt, @adjustable_tongs) #yaml setting true AND tongs actually adjustable
+          @command = "push fuel with my tongs"
+          @item = "tongs"
+          @stow = true
+        elsif @adjustable_tongs #yaml setting incorrect
```

Comment:
Would it be easier to check if the tongs are adjustable at script startup once? That way we know 1) if they are adjustable; and 2) what their current state is - and the loop can just use that known state.

Reply:
1, doesnt always call for shovel, so startup status check isnt necrssary, 2, if smelt/forge/etc is called multiple times, eg smith, weve now got startup checks every time it runs, rather than as needed for the first shovel call.